components/brave_wallet/browser/json_rpc_requests_helper.h

```diff
@@ -7,36 +7,38 @@
 #define BRAVE_COMPONENTS_BRAVE_WALLET_BROWSER_JSON_RPC_REQUESTS_HELPER_H_
 
 #include <string>
+#include <utility>
+
 #include "base/values.h"
 
 // Helper functions for building out JSON RPC requests across all blockchains.
 namespace brave_wallet {
 
-base::Value GetJsonRpcDictionary(const std::string& method,
-                                 base::Value* params);
-
-std::string GetJSON(const base::Value& dictionary);
+namespace internal {
 
-std::string GetJsonRpcNoParams(const std::string& method);
+base::Value::Dict ComposeRpcDict(const base::StringPiece method);
 
-std::string GetJsonRpc1Param(const std::string& method, const std::string& val);
+}
 
-std::string GetJsonRpc2Params(const std::string& method,
-                              base::Value&& val1,
-                              base::Value&& val2);
+template <typename T>
+base::Value::Dict GetJsonRpcDictionary(base::StringPiece method, T params) {
+  auto dict = internal::ComposeRpcDict(method);
+  dict.Set("params", std::move(params));
+  return dict;
+}
 
-std::string GetJsonRpc2Params(const std::string& method,
-                              const std::string& val1,
-                              const std::string& val2);
+std::string GetJSON(base::ValueView dict);
 
-std::string GetJsonRpc3Params(const std::string& method,
-                              const std::string& val1,
-                              const std::string& val2,
-                              const std::string& val3);
+template <typename... Args>
+std::string GetJsonRpcString(base::StringPiece method, Args... args) {
+  base::Value::List params;
+  (params.Append(std::move(args)), ...);
+  return GetJSON(GetJsonRpcDictionary(method, std::move(params)));
+}
 
-void AddKeyIfNotEmpty(base::Value* dict,
-                      const std::string& name,
-                      const std::string& val);
+void AddKeyIfNotEmpty(base::Value::Dict* dict,
+                      base::StringPiece name,
+                      base::StringPiece val);
```

Comment:
Actually we are supposed to pass const by value, as the two members can fit on a register.

Reply:
I see. then we need to change the definition here without a const since it is a view
```
void AddKeyIfNotEmpty(base::Value::Dict* dict,
                      const base::StringPiece name,
                      const base::StringPiece val) {
```

Reply:
Fixed